testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -5847,3 +5873,199 @@ fn atlas_stress_integration_test() {
+            &max_contract_src,
+        ),
+    );
+    run_until_burnchain_height(&mut btc_regtest_controller, &blocks_processed, 212, &conf);
+
+    // Loop 10 times. Each time, execute the same transaction, but increase the amount *paid*.
+    // We will then check that the fee rates always go up, but the cost stays the same.
+    let mut response_estimated_costs = vec![];
+    let mut response_top_fee_rates = vec![];
+    for i in 1..11 {
+        submit_tx(
+            &http_origin,
+            &make_contract_call(
+                &spender_sk,
+                i,          // nonce
+                i * 100000, // payment
+                &spender_addr.into(),
+                "increment-contract",
+                "increment-many",
+                &[],
+            ),
+        );
+        run_until_burnchain_height(
+            &mut btc_regtest_controller,
+            &blocks_processed,
+            212 + 2 * i,
+            &conf,
+        );
+
+        {
+            // Read from the fee estimation endpoin.
+            let path = format!("{}/v2/fees/transaction", &http_origin);
+
+            let contract_addr = to_addr(&StacksPrivateKey::from_hex(SK_1).unwrap());
+            let tx_payload = TransactionPayload::ContractCall(TransactionContractCall {
+                address: contract_addr.clone().into(),
+                contract_name: ContractName::try_from("increment-contract").unwrap(),
+                function_name: ClarityName::try_from("increment-many").unwrap(),
+                function_args: vec![],
+            });
+
+            let payload_data = tx_payload.serialize_to_vec();
+            let payload_hex = format!("0x{}", to_hex(&payload_data));
+
+            let body = json!({ "transaction_payload": payload_hex.clone() });
+
+            let client = reqwest::blocking::Client::new();
+            let fee_rate_result = client
+                .post(&path)
+                .json(&body)
+                .send()
+                .expect("Should be able to post")
+                // .json::<serde_json::Value>()
+                .json::<RPCFeeEstimateResponse>()
+                .expect("Failed to parse result into JSON");
+
+            response_estimated_costs.push(fee_rate_result.estimated_cost_scalar);
+            response_top_fee_rates.push(fee_rate_result.estimations.last().unwrap().fee_rate);
+        }
+    }
+
+    assert_eq!(response_estimated_costs.len(), response_top_fee_rates.len());
+
+    // Check that:
+    // 1) The cost is always the same.
+    // 2) The estimated fee rate only either i) moves up, ii) stays within random noise of where it
+    //    was.
+    for i in 1..response_estimated_costs.len() {
+        let curr_rate = response_top_fee_rates[i] as f64;
+        let last_rate = response_top_fee_rates[i - 1] as f64;
+        // Check that either 1) curr_rate is bigger, or 2) is only smaller by random 11% noise.
+        assert!(curr_rate - last_rate > -1.0 * curr_rate * 0.11);
```

Comment:
I'm a little confused by the reasoning behind this assertion - could you explain?

Reply:
This comment has been addressed in the latest revision, by using the new config parameters to simplify the test.